### PR TITLE
Revit file sharing

### PR DESCRIPTION
**Pull request by adamhathcock**
Overrides https://github.com/specklesystems/speckle-sharp/pull/3468

Uses info from https://github.com/dotnet/sdk/issues/2511

Basically, shared projects are deprecated and only made because of UWP.  Using the new build system features makes us more future proof

### CODE REVIEW

File: DUI3-DX/Connectors/Revit/Speckle.Connectors.RevitShared/Speckle.Connectors.RevitShared.csproj
Comment (BovineOx) [inline DUI3-DX/Connectors/Revit/Speckle.Connectors.RevitShared/Speckle.Connectors.RevitShared.csproj:1]: This is what makes it shared, I guess?

File: DUI3-DX/Converters/Revit/Speckle.Converters.RevitShared/Raw/Geometry/SolidConversionToSpeckle.cs
Comment (BovineOx) [inline DUI3-DX/Converters/Revit/Speckle.Converters.RevitShared/Raw/Geometry/SolidConversionToSpeckle.cs:45]: Yeah.... much better

**Review by BovineOx — APPROVED** [review]
I think it's fine but @AlanRynne should review also before merging and also we must eb sure it's all good on the CI

File: DUI3-DX/Connectors/Revit/Speckle.Connectors.Revit2023/Speckle.Connectors.Revit2023.csproj
Comment (AlanRynne) [inline DUI3-DX/Connectors/Revit/Speckle.Connectors.Revit2023/Speckle.Connectors.Revit2023.csproj:37]: I've never had to run this Exclude attribute for the `obj` folder. 🤔

It kinda feels wrong

File: DUI3-DX/Connectors/Revit/Speckle.Connectors.RevitShared/Speckle.Connectors.RevitShared.csproj
Comment (AlanRynne) [inline DUI3-DX/Connectors/Revit/Speckle.Connectors.RevitShared/Speckle.Connectors.RevitShared.csproj:15]: If the .xaml and .cs files live in this project, why do we need to also add them to the specific 2023 project as well?

Also, that exclude for the `obj` folder feels wrong 😅

**Review by AlanRynne — COMMENTED** [review]
Got some questions about this before we merge. My understanding is that `NoTarget` SDK projects are intended for projects that have no code but do have project dependencies (i.e. building it results in no DLL being built, just it's dependencies being copied)
The way we're doing this is somewhat opposite to that 🤔

File: DUI3-DX/Connectors/Revit/Speckle.Connectors.RevitShared/Speckle.Connectors.RevitShared.csproj
Comment (adamhathcock) [reply DUI3-DX/Connectors/Revit/Speckle.Connectors.RevitShared/Speckle.Connectors.RevitShared.csproj:1]: This csproj isn't used at all so we don't want msbuild to do anything but can be seen in the sln

File: DUI3-DX/Connectors/Revit/Speckle.Connectors.Revit2023/Speckle.Connectors.Revit2023.csproj
Comment (adamhathcock) [reply DUI3-DX/Connectors/Revit/Speckle.Connectors.Revit2023/Speckle.Connectors.Revit2023.csproj:37]: making a csproj just for shared makes an obj folder.

we don't strictly need the csproj for shared (since each individual "real" csproj includes the files separately) but it's making the shared stuff viewable in the sln

File: DUI3-DX/Connectors/Revit/Speckle.Connectors.RevitShared/Speckle.Connectors.RevitShared.csproj
Comment (adamhathcock) [reply DUI3-DX/Connectors/Revit/Speckle.Connectors.RevitShared/Speckle.Connectors.RevitShared.csproj:15]: they don't really "live" here....it's not used by other csprojs if you look

**Comment by adamhathcock** [comment]
> Got some questions about this before we merge. My understanding is that `NoTarget` SDK projects are intended for projects that have no code but do have project dependencies (i.e. building it results in no DLL being built, just it's dependencies being copied) The way we're doing this is somewhat opposite to that 🤔

You're right....the shared csprojs could be deleted and we wouldn't need to exclude the OBJ folder as it wouldn't exist.  The shared csproj is just for viewing in the SLN

files are directly added to each csproj via the compile include

**Comment by adamhathcock** [comment]
went through this again with @AlanRynne in some detail and actually gives a slightly worse DX even though shprojs aren't supported going forward.  Taking this on the chin for now.

**Pull request closed without merging**
